ops/main.py

```diff
@@ -276,10 +276,50 @@ def is_restricted_context(self):
         return self.event_name in ('collect_metrics',)
 
 
-def main(charm_class, use_juju_for_storage=False):
+def _should_use_controller_storage(db_path: Path, meta: ops.charm.CharmMeta) -> bool:
+    """Figure out whether we want to use controller storag or not."""
+    # if you've previously used local state, carry on using that
+    if db_path.exists():
+        logger.debug("Using local storage: %s already exists", db_path)
+        return False
+
+    # if you're not in k8s you don't need controller storage
+    if 'kubernetes' not in meta.series:
+        logger.debug("Using local storage: not a kubernetes charm")
+        return False
+
+    # if the charm specifies a min juju version, use that
+    if meta.min_juju_version is not None:
+        min_version = JujuVersion(meta.min_juju_version)
+        if min_version.has_controller_storage():
+            logger.debug("Using controller storage: min_juju_version: %s", min_version)
+            return True
+        else:
+            logger.debug("Using local storage: min_juju_version: %s", min_version)
+            return False
+
+    # min_juju_version not set, check current juju version
+    cur_version = JujuVersion.from_environ()
+
+    if cur_version.has_controller_storage():
+        logger.debug("Using controller storage: JUJU_VERSION=%s", cur_version)
```

Comment:
Question (may want to improve consistency, but I'm more interested in the reason): Why above (l.295) you do `foo: %s` and here `foo=%s`?

Reply:
I'm mimicking how that thing being reported on is set / used; the `foo: %s` comes from YAML, while this `FOO=%s` comes form an env var